pentaho-requirejs-osgi-manager/core/impl/src/main/java/org/pentaho/requirejs/impl/servlet/RequireJsConfigServlet.java

```diff
@@ -214,7 +214,12 @@ private String getRequireJsScript() throws IOException {
       this.outputRequireJs = this.getBooleanValue( req.getParameter( "requirejs" ), true );
 
       this.referer = req.getHeader( "referer" );
-      this.serverAddress = req.getScheme() + "://" + req.getServerName() + ":" + req.getServerPort();
+
+      // To be congruent with referer other popular http clients the port should be stripped from the 'Host' field
+      // when the port is 80.
+      this.serverAddress = ( 80 == req.getServerPort() || 443 == req.getServerPort() )
```

Comment:
can we have final variables for 80 and 443 ? the variable should have a clear name that indicates that these are the default ports.